### PR TITLE
feat(site): add template insights page

### LINKED ISSUE #8689: Feature: Insights page for Coder
Per [RFC](https://github.com/coder/coder/discussions/8109), slack threads, and recent backend contributions: https://github.com/coder/coder/issues/8514, https://github.com/coder/coder/pull/8519, I guess it's the high time to implement site changes.

### PR DESCRIPTION
**Pull request by BrunoQuaresma**
Close https://github.com/coder/coder/issues/8689

Demo:

https://github.com/coder/coder/assets/3165839/10b0a595-451a-4b76-9896-cab805a0974f

Next steps:
- Add a date range picker for custom dates
- Add template parameters usage section

### CODE REVIEW

**Comment by mtojek** [comment]
@BrunoQuaresma could you please post some screenshots?

**Comment by BrunoQuaresma** [comment]
@mtojek I posted a video!

File: codersdk/deployment.go [hunk 1]
```diff
@@ -1862,7 +1865,9 @@ const (
 // users to opt-in to via --experimental='*'.
 // Experiments that are not ready for consumption by all users should
 // not be included here and will be essentially hidden.
-var ExperimentsAll = Experiments{}
+var ExperimentsAll = Experiments{
+	ExperimentTemplateInsightsPage,
```
Comment (mtojek) [inline codersdk/deployment.go:1869]: Is it intended to release as experimental?

File: site/src/components/DAUChart/DAUChart.test.tsx [hunk 2]
```diff
@@ -1,37 +0,0 @@
-import { render } from "testHelpers/renderHelpers"
```
Comment (mtojek) [inline site/src/components/DAUChart/DAUChart.test.tsx:1]: Is this deletion intended?

File: site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+    const remainingMinutes = Math.floor((seconds % 3600) / 60)
+    if (remainingMinutes === 0) {
+      return hours + " hours"
+    } else {
+      return hours + " hours, " + remainingMinutes + " minutes"
+    }
+  }
+}
+
+function sevenDaysAgo() {
```
Comment (mtojek) [inline site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx:358]: nit: weekAgo :)

File: site/yarn.lock [hunk 4]
```diff
@@ -3506,6 +3506,11 @@
     "@types/connect" "*"
     "@types/node" "*"
 
+"@types/chroma-js@2.4.0":
```
Comment (mtojek) [inline site/yarn.lock:3509]: Is this change intended?

**Review by mtojek — APPROVED** [review]
Looks good to me 👍 I played a bit with this, and seems to work correctly.

Could you please write in the PR description what are the next steps? I assume that this is the MVP.

File: codersdk/deployment.go (hunk 1 above)
Comment (BrunoQuaresma) [reply codersdk/deployment.go:1869]: Yes, I talked with @bpmct about this

File: site/src/components/DAUChart/DAUChart.test.tsx (hunk 2 above)
Comment (BrunoQuaresma) [reply site/src/components/DAUChart/DAUChart.test.tsx:1]: Yeap, it is just a render test which should be on the storybook.

File: site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx (hunk 3 above)
Comment (BrunoQuaresma) [reply site/src/pages/TemplatePage/TemplateInsightsPage/TemplateInsightsPage.tsx:358]: This is a good one. I don't think a "week ago" is a good name for this because you are at Wed and a week ago would be the previous week from Mon to Fri so 7 days ago is different and for this case would be more accurate

File: site/yarn.lock (hunk 4 above)
Comment (BrunoQuaresma) [reply site/yarn.lock:3509]: it is not! good catch

Comment (BrunoQuaresma) [reply site/yarn.lock:3509]: yes, it is!
